### DISCUSSION TITLE
Library for Solar Production Forecast (PV Power)

### DISCUSSION THREAD

**Discussion by rany2** [open-meteo/open-meteo#813] [Show and tell]
I've been using this for the past two weeks and it seems to be working perfectly fine. In fact, I was really surprised when I realized it works better than commercial offerings (Solcast and Forecast.Solar).

The library is available here: https://github.com/rany2/open-meteo-solar-forecast
HA integration here: https://github.com/rany2/ha-open-meteo-solar-forecast

Thanks a lot for this great project, initially I tried to make use of the raw data downloaded from re.jrc.ec.europa.eu and it worked "okay-ish" but Open-Meteo was just better and I was able to take ambient temperature into account in my PPV calculations.

**Comment by patrick-zippenfenig** [comment]
Thanks for sharing! Please note that although you get good results for solar radiation out-of-the box, I would still recommend to calibrate the forecast based on measurements and use machine learning approaches.

In the future, I would like to add satellite solar radiation observations which could be very helpful to compare solar forecasts.

**Reply by rany2** [reply]
Thanks for your reply, that would be an interesting next step. Some commercial offerings allow users to upload their actual output to feed it into some ML system but I'll have to do some research to see how I could get started with it; I'm a beginner at this and all this is very much above my skill level.

Right now I'm struggling with compensating for cloud cover. I'm not sure if an ML system would help me (or how it could help me) with cloud cover because the data I get from Open-Meteo doesn't seem to account for cloud cover properly during the winter (I'm using global_tilted_irradiance).

I tried to slightly modify the cloud factor formula by Kasten-Czeplak to try and get the data to fit my winter data but it just seemed too hacky so I ditched it all together; especially given that Open-Meteo's GTI already accounts for cloud cover. The trick did work somewhat though but on the whole it was too pessimistic.

The only reassuring thing is that it seems like some luck is involved because at the moment when Open-Meteo nails the output during the winter, Solcast is off or vice versa. So there's definitely some chance involved and it's not an exact science.

**Edit:** Small update: it seems like only using cloud_cover_low with Kasten-Czeplak works well.
